cpp/cograph_recognition/cograph/Recognition.cpp

```diff
@@ -0,0 +1,233 @@
+#include <recognition/CographRecognition.hpp>
+#include <list>
+#include <set>
+
+#include <graph/GraphTools.hpp>
+namespace Koala {
+enum class Type{
+ZEROONE,
+VERTEX
+};
+enum class Marked{
+    UNMARKED,
+    MARKED,
+    MARKED_AND_UNMARKED
+}
+class CoNode{
+    private:
+    Type type;
+    int number;
+    Marked marked;
+    int md, d;
+    //md is the current number of children, which have been both "marked" and "unmarked"
+    //d is the current number of children
+
+    CoNode* head_of_list_of_children;
+    CoNode *next, *prev;//in list of children of its parent
+    CoNode *parent;
+    vector<CoNode>outEdges;//neighbours of cur vertex in G
+    public:
+    CoNode(Type type, int number): type(type), number(number),marked(Marked::UNMARKED),md(0),d(0),head_of_list_of_children(nullptr),
+    next(nullptr),prev(nullptr), parent(nullptr),outEdges(vector<CoNode>(0)){
+
+    }
+    void addchild(CoNode *x){
```

Comment:
Please stick to the convention: public function and classes have SuchAndSuchNames, private such_and_such_names

And please fix the indentation or use a linter, it's more than aesthetics, it really helps reading.